### DISCUSSION TITLE
Deploying a solution with pg gives error: Could not resolve "pg-native"

### DISCUSSION THREAD

**Discussion by jesperordrup** [vercel/community#228] [Help] [answered]
Im using pg in a few endpoints. 

I get this error when I deploy a sveltekit app to vercel. Error taken from log.
![image](https://user-images.githubusercontent.com/3120151/150689851-d80d230f-1e14-40d6-974f-4aee896577d7.png)

Am I supposed to do any further configuration?

**Comment by leerob** [comment]
I would recommend using Postgres through an API, to prevent having to open a connection to the database on every request. A great example of this would be Supabase.

More details: https://twitter.com/leeerob/status/1430177168307412998

(Mentioning this because you wouldn't need `pg` and instead would use `fetch` or their native ORM / library)

**Reply by jesperordrup** [reply]
I understand the connect/pool/maxing out of connection problem. 

I want to use all the cool db's out there but im forced to use a selfhosted db.

So basically in serverless: dont do db's which uses clients that requires a connection directly from serverless. Which means dont do mongodb, postgress, mssql unless adding middleman (server or other service) to handle pools/connections. 

Simplicity takes a big hit as it surely adds a server (or service) to the serverless setup right?

Dreaming: If only there were some way to persist a variable in a serverless setup then all would be solved?

ps. tried to find how you solved this for mongodb but https://vercel.com/integrations#databases - all links to databases are 404

**Reply by leerob** [reply]
> Which means dont do mongodb, postgress, mssql unless adding middleman (server or other service) to handle pools/connections.

You can absolutely still do these, they just require thinking about the architecture a bit differently in a serverless, auto-scaling world than with a persistent, always-on server. Many customers are using these databases with their frontend on Vercel.

> Simplicity takes a big hit as it surely adds a server (or service) to the serverless setup right?

Not necessarily! For example, [Mongo's Data API](https://docs.atlas.mongodb.com/api/data-api/) makes this really easy.

Note: we fixed the 404 bug, pushed some bad code earlier that was patched up 😄 #whoops

**Comment by jesperordrup** [comment]
Thank you. 

So my original question about pg-native error - is that vercel related? 

Regarding my dreaming, could you help me out understanding why it's not so brilliant?

**Reply by leerob** [reply]
Not Next.js or Vercel specific, no: https://github.com/serverless-heaven/serverless-webpack/issues/78

**Comment by jesperordrup** [comment, marked as answer]
Thanks @leerob 

As soon as I see webpack I turn around :-) as earlier I've never felt confident enough to do anything other than copy and paste webpack stuff.

Btw it's actually a sveltekit solution (tried both the auto and vercel adapter).

Im sure it's not vercel but since I chose vercel to make this deploy easy I've pivoted to deploying to a dokku container.   This actually  deployed without any problems  (with adapter-node) - making it simple for me again. So maybe that's best for these kind of solutions I guess.
